src/annotations/AnnotationManager.php

```diff
@@ -299,6 +306,22 @@ protected function getAnnotations($class_name, $member_type = self::MEMBER_CLASS
                 }
             }
 
+            if ($this->_traitsSupported && $member_name) {
+                $traitAnnotations = array();
+
+                if (isset($file->traitMethodOverrides[$class_name][$member_name])) {
+                    list($traitName, $originalMemberName) = $file->traitMethodOverrides[$class_name][$member_name];
+                    $traitAnnotations = $this->getAnnotations($traitName, $member_type, $originalMemberName);
+                } else {
+                    foreach ($reflection->getTraitNames() as $traitName)
+                    {
+                        $traitAnnotations += $this->getAnnotations($traitName, $member_type, $member_name);
+                    }
+                }
+
+                $annotations = array_merge($traitAnnotations, $annotations);
```

Comment:
I'm not talking about case you're described (trait method replaces class method inherited from its parent class), but rather about case when trait method has no effect, when method with same name is defined in same class, that has "use" statement for that trait.